### PR TITLE
* Use the presence of hash file to denote completion of package install

### PR DESCRIPTION
**Pull request by pranavkm**
operation removing the need for Directory.Move.

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl @ChengTian 

@BrennanConroy, I tried this on your VM and it does look like it's resolved correctly this time around.

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -414,7 +416,7 @@ public RestoreCommand(IApplicationEnvironment env)
                 }
             });
 
-            await InstallPackages(installItems, packagesDirectory, packageFilter: (library, nupkgSHA) => true);
```
Comment (pranavkm) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Unless I'm missing something super obvious, `packageFilter` never did anything particularly useful.

**Comment by ChengTian** [comment]
I can help to verify this change on my super slow Win7 VM. Will post feedback here after that.

**Comment by ChengTian** [comment]
Being blocked by an issue on Win7: https://github.com/aspnet/dnx/issues/1616
I cannot do the verification temporarily.

**Comment by ChengTian** [comment]
Wrote a script and ran 1000+ times of `dnu restore --parallel` for `MusicStore` on a very slow Win7 VM. Couldn't reproduce the bug. Looks like it does fix the bug.

**Comment by pranavkm** [comment]
Great. Could you review the PR so we can check it in and start using it as part of the CI build script?

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs (hunk 1 above)
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Yeah, I also removed this part during `dnu refactoring`

**Comment by davidfowl** [comment]
@pranavkm Why is the build is failing?

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 2]
```diff
@@ -635,30 +637,33 @@ internal bool IsBestVersion(string name, SemanticVersion version)
...
+            using (var memoryStream = new MemoryStream())
+            {
+                var match = installItem.Match;
+                await match.Provider.CopyToAsync(installItem.Match, memoryStream);
 
-                    Reports.Information.WriteLine("Installing {0} {1}", library.Name.Bold(), library.Version);
-                    memStream.Seek(0, SeekOrigin.Begin);
-                    await NuGetPackageUtils.InstallFromStream(memStream, library, packagesDirectory, sha512);
-                }
+                memoryStream.Position = 0;
```
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Why do we replace `memStream.Seek(0, SeekOrigin.Begin);` with this?

**Comment by pranavkm** [comment]
Travis seems to hung due to package restore - `No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.` AppVeyor doesn't have any build logs. Let me try rebasing and pushing to this branch - so we'll get a new build.

**Comment by davidfowl** [comment]
It's because you added --quiet. You might need to turn it back on.

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs (hunk 2 above)
Comment (pranavkm) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: I can revert this - but they're functionally identical.

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Yes, please revert it. I guess it's better to use the same approach consistently.

**Comment by pranavkm** [comment]
Updated.

**Comment by ChengTian** [comment]
:shipit:

**Pull request closed without merging**